File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        mapp = dict()
        mapp[0] = 1
        ct = 0
        cursum = 0
        for i in nums:
            cursum += i
            dif = cursum - k
            ct += mapp.get(dif, 0)
            mapp[cursum] = mapp.get(cursum, 0) + 1
        return ct

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        map<int,int> prefixes;
        prefixes[0] = 1;
        int ct = 0;
        int cursum = 0;
        for(int i = 0; i < nums.size(); i++){
            cursum += nums[i];
            int dif = cursum - k;
            ct += prefixes[dif];
            prefixes[cursum]++;
        }
        return ct;
    }
};